File: scripts/plot_v3_reads.R
#!/usr/bin/env R

setwd("/home/youri/projects/gsam")

d <- read.delim("output/tables/v3.txt",stringsAsFactors = F)
e <- d[!is.na(d$resection.1.wt) & !is.na(d$resection.2.wt),]


off <- 0.05
plot(c(1.0 - off,2.0 + off), c(0,100),type="n", ylab="%vIII",xlab="resection",main=paste0("vIII over resection/time in ",nrow(e)," matching samples"))

for(i in 1:nrow(e)) {
  p1 <- 100.0 *   e[i,4] / (e[i,3] + e[i,4])
  p2 <- 100.0 *   e[i,7] / (e[i,6] + e[i,7])
  
  points(1, p1, pch=19,cex=0.6)
  points(2, p2, pch=19,cex=0.6)
  lines(c(1,2), c(p1, p2))
  
  text(1,p1, e[i,1],cex=0.65,pos=2,col="darkgray")
  text(2,p2, e[i,1],cex=0.65,pos=4,col="darkgray")
}

File: scripts/plot_v3_reads.py
#!/usr/bin/env python

idx = {}

import os
from tqdm import tqdm

with open("output/tables/v3.txt", "w") as fh_out:
    fh_out.write("sid\t\tresection-1-wt\tresection-1-v3\t\tresection-2-wt\tresection-2-v3\n")

    with open("output/tables/v3_dedup.txt") as fh:
        for line in tqdm(fh):
            line = line.strip()
            if len(line) > 0:
                line = line.split("\t")
                if line[0] != "sample":
                    sid = line[0].split("Pfrench_")[1].split("_")[0]
                    resection = sid[3]
                    sid = sid[0:3]

                    if sid not in idx:
                        idx[sid] = {'1': ["NA", "NA"], '2': ["NA", "NA"]}

                    idx[sid][resection] = line[1:]


    for sid in sorted(idx.keys()):
        fh_out.write(sid + "\t|\t" + idx[sid]["1"][0] + "\t" + idx[sid]["1"][1] + "\t|\t" +  idx[sid]["2"][0] + "\t" + idx[sid]["2"][1] + "\n")

